### PR TITLE
Add helm timeout to avoid infinitive busy loop while reconciling helm extension

### LINKED ISSUE #1492: Helm extension spawn mutliple releases on failure
### Before creating an issue, make sure you've checked the following:

- [X] You are running the latest released version of k0s
- [X] Make sure you've searched for existing issues, both open and closed
- [X] Make sure you've searched for PRs too, a fix might've been merged already
- [X] You're looking at docs for the released version, "main" branch docs are usually ahead of released versions.

### Version

v1.23.3+k0s.0

### Platform

```shell
No LSB modules are available.
Distributor ID:	Debian
Description:	Debian GNU/Linux 11 (bullseye)
Release:	11
Codename:	bullseye
```

### What happened?

I added sentry helm chart via extension section in k0s.yaml. It is complex chart with bunch of components and subcharts. For some reason k0s failed to install chart and produce 10+ releases of this chart and eat all resources of my test single node cluster. I can send all systemd journals if it matter

### Steps to reproduce

k0s.yaml
```
spec:
  extensions:
    storage:
      type: openebs_local_storage
    helm:
      repositories:
      - name: bitnami
        url: https://charts.bitnami.com/bitnami
      - name: ingress-nginx
        url: https://kubernetes.github.io/ingress-nginx
      - name: jetstack
        url: https://charts.jetstack.io
      - name: sentry
        url: https://sentry-kubernetes.github.io/charts
      charts:
      - name: nginx
        namespace: ingress
        chartname: ingress-nginx/ingress-nginx
        version: "4.0.1"
        values: |
          [SKIPED]
      - name: postgres
        namespace: db
        chartname: bitnami/postgresql
        version: "10.16.2"
        values: |
          [SKIPED]
      - name: sentry
        namespace: sentry
        chartname: sentry/sentry
        version: "13.0.1"
        values: |
          #for first run
          asHook: true
          sentry:
            worker:
              replicas: 1
          mail:
            backend: smtp
            useTls: true
            username: "sentry"
            password: "xxxxxxxxxxxx"
            port: "25"
            host: "smtp.mail.svc"
            from: "noreply@xxxx.ru"
          auth:
            register: false
          ingress:
            enabled: "true"
            hostname: snt.xxxx.ru
            annotations:
              kubernetes.io/ingress.class: nginx
              nginx.ingress.kubernetes.io/use-regex: "true"
              nginx.ingress.kubernetes.io/force-ssl-redirect: 'true'
              cert-manager.io/cluster-issuer: 'letsencrypt-prod'
            tls:
            - hosts:
              - 'snt.xxxx.ru'
              secretName: snt.xxxx.ru-tls
          nginx:
            enabled: false
          postgresql:
            enabled: false
          externalPostgresql:
            host: postgres.db.svc
            password: xxx
          redis:
            cluster:
              enabled: false
            replica:
              replicaCount: 0
          rabbitmq:
            ## If disabled, Redis will be used instead as the broker.
            enabled: false
          kafka:
            replicaCount: 1
            defaultReplicationFactor: 1
            offsetsTopicReplicationFactor: 1
            transactionStateLogReplicationFactor: 1
            transactionStateLogMinIsr: 1
          clickhouse:
            clickhouse:
              replicas: "1"
            # Web interface for ClickHouse in the Tabix project
            tabix:
              enabled: false
      - name: cert-manager
        namespace: cert-manager
        chartname: jetstack/cert-manager
        version: "v1.7.0"
```

`k0s stop && k0s start`

### Expected behavior

One release of chart created; don't try to create again and again in case of failure

### Actual behavior

```
# w
 10:14:13 up 2 days, 22:55,  1 user,  load average: 90.60, 98.72, 105.46
```

```
# k0s kubectl -n ingress get pod -A | grep kafka
sentry         sentry-1643988150-kafka-0                                     1/1     Running                  0              14h
sentry         sentry-1643989158-kafka-0                                     1/1     Running                  0              14h
sentry         sentry-1643989167-kafka-0                                     1/1     Running                  0              14h
sentry         sentry-1643989196-kafka-0                                     0/1     Pending                  0              14h
sentry         sentry-1643989244-kafka-0                                     1/1     Running                  0              14h
sentry         sentry-1643989295-kafka-0                                     0/1     CreateContainerError     178            14h
sentry         sentry-1643989333-kafka-0                                     1/1     Running                  0              14h
sentry         sentry-1643989489-kafka-0                                     1/1     Running                  4 (14h ago)    14h
sentry         sentry-1643989504-kafka-0                                     1/1     Running                  3 (14h ago)    14h
sentry         sentry-1643990366-kafka-0                                     0/1     CreateContainerError     174            14h
sentry         sentry-1643991686-kafka-0                                     1/1     Running                  1 (13h ago)    13h
sentry         sentry-1643992982-kafka-0                                     0/1     Pending                  0              13h
```

Since there is cronjobs and other all those releases spawn too many pods:
```
sentry         sentry-1643989333-snuba-cleanup-errors-27400380-78hg5         0/1     OutOfpods                0             3h50m
sentry         sentry-1643989333-snuba-cleanup-errors-27400380-7dkzt         0/1     Pending                  0             3h14m
sentry         sentry-1643989333-snuba-cleanup-errors-27400380-95s8g         0/1     ContainerStatusUnknown   1             4h52m
sentry         sentry-1643989333-snuba-cleanup-errors-27400380-n227d         0/1     OutOfpods                0             3h49m
sentry         sentry-1643989333-snuba-cleanup-errors-27400440-2p9ns         0/1     OutOfpods                0             3h53m
sentry         sentry-1643989333-snuba-cleanup-errors-27400440-5xt7s         0/1     Pending                  0             74m
sentry         sentry-1643989333-snuba-cleanup-errors-27400440-8w56t         0/1     OutOfpods                0             75m
sentry         sentry-1643989333-snuba-cleanup-errors-27400440-lnlnb         0/1     OutOfpods                0             75m
```

### Screenshots and logs

```
k0s kubectl get chart -n kube-system 

NAME                           AGE
k0s-addon-chart-cert-manager   15h
k0s-addon-chart-nginx          2d22h
k0s-addon-chart-openebs        2d20h
k0s-addon-chart-postgres       23h
k0s-addon-chart-sentry         15h
```

```
# helm list -A
WARNING: Kubernetes configuration file is group-readable. This is insecure. Location: /var/lib/k0s/pki/admin.conf
NAME                   	NAMESPACE   	REVISION	UPDATED                                	STATUS  	CHART              	APP VERSION
cert-manager-1643989551	cert-manager	1       	2022-02-04 18:46:25.72661292 +0300 MSK 	failed  	cert-manager-v1.7.0	v1.7.0     
cert-manager-1643990607	cert-manager	1       	2022-02-04 19:04:22.32242738 +0300 MSK 	failed  	cert-manager-v1.7.0	v1.7.0     
cert-manager-1643991800	cert-manager	1       	2022-02-04 19:24:41.513367817 +0300 MSK	failed  	cert-manager-v1.7.0	v1.7.0     
cert-manager-1643993119	cert-manager	1       	2022-02-04 19:46:52.674129093 +0300 MSK	failed  	cert-manager-v1.7.0	v1.7.0     
openebs-1643797942     	openebs     	36      	2022-02-04 18:44:34.943249938 +0300 MSK	deployed	openebs-3.0.3      	3.0.1      
postgresql-1643958937  	db          	9       	2022-02-04 18:44:47.940129632 +0300 MSK	deployed	postgresql-10.16.2 	11.14.0    
sentry-1643988150      	sentry      	1       	2022-02-04 18:22:34.853427437 +0300 MSK	failed  	sentry-13.0.1      	21.8.0     
sentry-1643989158      	sentry      	1       	2022-02-04 18:39:23.273735984 +0300 MSK	failed  	sentry-13.0.1      	21.8.0     
sentry-1643989167      	sentry      	1       	2022-02-04 18:39:47.918244977 +0300 MSK	failed  	sentry-13.0.1      	21.8.0     
sentry-1643989196      	sentry      	1       	2022-02-04 18:40:29.653524604 +0300 MSK	failed  	sentry-13.0.1      	21.8.0     
sentry-1643989244      	sentry      	1       	2022-02-04 18:41:17.502844616 +0300 MSK	failed  	sentry-13.0.1      	21.8.0     
sentry-1643989295      	sentry      	1       	2022-02-04 18:41:59.501190616 +0300 MSK	failed  	sentry-13.0.1      	21.8.0     
sentry-1643989333      	sentry      	1       	2022-02-04 18:42:45.90448219 +0300 MSK 	failed  	sentry-13.0.1      	21.8.0     
sentry-1643989489      	sentry      	1       	2022-02-04 18:44:59.221574178 +0300 MSK	failed  	sentry-13.0.1      	21.8.0     
sentry-1643989504      	sentry      	1       	2022-02-04 18:45:30.862394851 +0300 MSK	failed  	sentry-13.0.1      	21.8.0     
sentry-1643990366      	sentry      	1       	2022-02-04 19:00:38.457259617 +0300 MSK	failed  	sentry-13.0.1      	21.8.0     
sentry-1643991686      	sentry      	1       	2022-02-04 19:22:48.499996776 +0300 MSK	failed  	sentry-13.0.1      	21.8.0     
sentry-1643992982      	sentry      	1       	2022-02-04 19:43:38.451402757 +0300 MSK	failed  	sentry-13.0.1      	21.8.0     
```

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by soider**
Signed-off-by: Mikhail Sakhnov <msakhnov@mirantis.com>

## Description

This PR introduces timeouts for the helm release handling to avoid infinitive locks in case if the release is totally broken and never finishes.

Partially fixes #1492 (other part was introduced in the #1461)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## How Has This Been Tested?

- [x] Manual test
- [x] Auto test added

## Checklist:

- [x] My code follows the style [guidelines](https://github.com/k0sproject/k0s/blob/main/docs/contributors/overview.md) of this project 
- [x] My commit messages are [signed-off](https://github.com/k0sproject/k0s/blob/main/docs/contributors/github_workflow.md)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

File: pkg/helm/helm.go [hunk 1]
```diff
@@ -208,25 +195,30 @@ func (hc *Commands) isInstallable(chart *chart.Chart) bool {
...
 
-func (hc *Commands) InstallChart(chartName string, version string, releaseName string, namespace string, values map[string]interface{}) (*release.Release, error) {
+func (hc *Commands) InstallChart(chartName string, version string, releaseName string, namespace string, values map[string]interface{}, timeout time.Duration) (*release.Release, error) {
 	cfg, err := hc.getActionCfg(namespace)
 	if err != nil {
 		return nil, fmt.Errorf("can't create action configuration: %v", err)
 	}
 	install := action.NewInstall(cfg)
 	install.CreateNamespace = true
+	install.WaitForJobs = true
```
Comment (makhov) [inline pkg/helm/helm.go:205]: Maybe we can skip it in case the timeout is 0.

**Review by makhov — COMMENTED** [review]
It would be good to update docs as well

File: pkg/component/controller/extensions_controller.go [hunk 2]
```diff
@@ -200,16 +201,29 @@ func (cr *ChartReconciler) uninstall(ctx context.Context, chart v1beta1.Chart) e
...
 }
 
+const defaultTimeout = time.Duration(10 * time.Minute)
+
 func (cr *ChartReconciler) updateOrInstallChart(ctx context.Context, chart v1beta1.Chart) error {
 	var err error
 	var chartRelease *release.Release
+	timeout, err := time.ParseDuration(chart.Spec.Timeout)
+	if err != nil || timeout == 0 {
+		cr.L.Tracef("Can't parse `%s` as time.Duration or got 0 value, using default timeout `%s`", chart.Spec.Timeout, defaultTimeout)
```
[the remark this answers is not in the document]
Comment (soider) [reply pkg/component/controller/extensions_controller.go]: done

File: inttest/addons/addons_test.go [hunk 3]
```diff
@@ -45,21 +45,22 @@ func (as *AddonsSuite) TestHelmBasedAddons() {
 	as.PutFile(as.ControllerNode(0), "/tmp/k0s.yaml", fmt.Sprintf(k0sConfigWithAddon, addonName))
 
 	as.Require().NoError(as.InitController(0, "--config=/tmp/k0s.yaml"))
-	as.waitForPrometheusRelease(addonName, 1)
+	as.NoError(as.RunWorkers())
+	kc, err := as.KubeClient(as.ControllerNode(0))
+	as.NoError(err)
```
Comment (twz123) [inline inttest/addons/addons_test.go:50]:
```suggestion
	as.Require().NoError(err)
```

**Review by makhov — APPROVED** [review]

**Pull request merged**